R/logging.R

```diff
@@ -307,8 +307,10 @@ log_problems <- function(notes, control, split, loc, res, bad_only = FALSE) {
     wrn_msg <- unique(wrn_msg)
     wrn_msg <- paste(wrn_msg, collapse = ", ")
 
+    wrn_traces <- purrr::map(wrn, `$`, "trace")
+
     wrn_msg <- tibble::new_tibble(
-      list(location = loc, type = "warning", note = wrn_msg),
+      list(location = loc, type = "warning", note = wrn_msg, trace = list(wrn_traces[[1]])),
```

Comment:
Note that traces are a tibble subclass.

Each row in the output of this function is a given place in evaluating a resample fold, e.g. `Fold1 preprocessor 1/1, model 1/1 (predictions)`.

It's possible to have more than one warning appear in a given model fit or prediction or metric calculation, so we collapse messages (see above) into one string in that case. For traces, we _could_ just pass all of `list(wrn_traces)`, but that would mean that, to be type-stable, we'd have to `list(list(err_cnd$trace))` for errors. 

Said another way, right now, to extract the second logged trace, you could write `collect_notes()$trace[[2]]`. If we supplied all of the warning traces in this line and changed the format for errors to be type-stable, you'd have to write `collect_notes()$trace[[2]][[1]]`. This `[[`-fu feels contrary to the point of the `collect_*()` functions. 

This approach banks on the fact that it's likely not useful to have a backtrace for each warning that occurs in a single call to e.g. `predict.xgb()`; the first is sufficient to help you drop into the right environment to track down an issue.

Reply:
I'd go with clarity over completeness here and only use the first trace as suggested. If we later realize that this does not get us to a useful debugging point often enough, we can still extend it.